src/controllers/beneficial.owner.other.controller.ts

```diff
@@ -1,29 +1,49 @@
 import { NextFunction, Request, Response } from "express";
-import { BENEFICIAL_OWNER_OTHER_PAGE, BENEFICIAL_OWNER_TYPE_URL } from "../config";
+import {
+  BENEFICIAL_OWNER_OTHER_PAGE,
+  BENEFICIAL_OWNER_TYPE_URL,
+  FEATURE_FLAG_ENABLE_REDIS_REMOVAL,
+  BENEFICIAL_OWNER_TYPE_WITH_PARAMS_URL,
+  ROUTE_PARAM_TRANSACTION_ID,
+  ROUTE_PARAM_SUBMISSION_ID
+} from "../config";
+import { isActiveFeature } from "../utils/feature.flag";
 import {
   getBeneficialOwnerOther,
   getBeneficialOwnerOtherById,
   postBeneficialOwnerOther,
   removeBeneficialOwnerOther,
   updateBeneficialOwnerOther
 } from "../utils/beneficial.owner.other";
+import { getUrlWithParamsToPath } from "../utils/url";
 
 export const get = (req: Request, res: Response) => {
-  getBeneficialOwnerOther(req, res, BENEFICIAL_OWNER_OTHER_PAGE, BENEFICIAL_OWNER_TYPE_URL);
+  getBeneficialOwnerOther(req, res, BENEFICIAL_OWNER_OTHER_PAGE, nextOrLastPage(req));
 };
 
 export const getById = (req: Request, res: Response, next: NextFunction) => {
-  getBeneficialOwnerOtherById(req, res, next, BENEFICIAL_OWNER_OTHER_PAGE, BENEFICIAL_OWNER_TYPE_URL);
+  getBeneficialOwnerOtherById(req, res, next, BENEFICIAL_OWNER_OTHER_PAGE, nextOrLastPage(req));
 };
 
 export const post = (req: Request, res: Response, next: NextFunction) => {
-  postBeneficialOwnerOther(req, res, next, BENEFICIAL_OWNER_TYPE_URL, true);
+  postBeneficialOwnerOther(req, res, next, nextOrLastPage(req), true);
 };
 
 export const update = (req: Request, res: Response, next: NextFunction) => {
-  updateBeneficialOwnerOther(req, res, next, BENEFICIAL_OWNER_TYPE_URL, true);
+  updateBeneficialOwnerOther(req, res, next, nextOrLastPage(req), true);
 };
 
 export const remove = (req: Request, res: Response, next: NextFunction) => {
-  removeBeneficialOwnerOther(req, res, next, BENEFICIAL_OWNER_TYPE_URL, true);
+  removeBeneficialOwnerOther(req, res, next, nextOrLastPage(req), true);
+};
+
+const nextOrLastPage = (req: Request): string => {
```

Comment:
Wondering why this function is named `nextOrLastPage`? I did a similar thing in my [PR](https://github.com/companieshouse/overseas-entities-web/pull/1145) and named it `getNextPageUrl`.

Reply:
It is used by the get methods and the arguments there are for the previous page, so covered both with the name.

Reply:
I think it's simply the next page URL that the app is going to redirect to. But up to you.

Reply:
Keep as is, if its a backlink next page could be misleading.